### PR TITLE
Add openai app example

### PR DESCRIPTION
**Pull request by horeaporutiu**
Add AI Image generation app to app-examples
* README w/ demo and template
* addressed Kirill's feedback

### CODE REVIEW

File: examples/ai-image-generator/pages/api/openai.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY, // This is also the default, can be omitted
+});
+
+export default async (req, res) => {
+  // grab prompt from the front end
+  let prompt = await req.body.prompt;
+  console.log("hello");
+  console.log(prompt);
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/api/openai.js:11]: remove?

File: examples/ai-image-generator/pages/index.jsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    };
+  }
+
+  const api = miro.as("");
+
+  const boards = [];
+
+  for await (const board of api.getAllBoards()) {
+    boards.push(board.name || "");
+  }
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/index.jsx:25]: As I understand, there is no need to load boards

File: examples/ai-image-generator/src/assets/congratulations.png
Comment (kirillsud) [inline examples/ai-image-generator/src/assets/congratulations.png:1]: Remove?

File: examples/ai-image-generator/src/assets/welcome.png
Comment (kirillsud) [inline examples/ai-image-generator/src/assets/welcome.png:1]: Remove?

File: examples/ai-image-generator/pages/index.jsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        url: target.src,
+      });
+      await window.miro.board.viewport.zoomTo(image);
+    }
+  };
+
+  const [inputValue, setInputValue] = useState("");
+
+  //handles the response being returned from backend (URL of image)
+  const [response, setResponse] = useState("");
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/index.jsx:69]: Remove?

File: examples/ai-image-generator/pages/index.jsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  function showSpinner() {
+    document.querySelector(".spinner").classList.add("show");
+  }
+
+  // Removes spinner when API calls are finished and data is returned
+  function removeSpinner() {
+    document.querySelector(".spinner").classList.remove("show");
+  }
+
+  const handleButtonClick = async () => {
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/index.jsx:86]: would be nice to see the same code style for all functions in the example. Currently, some of them are declared as a function (`function removeSpinner() {`), but some as a variable: `const handleButtonClick = async () => {`

**Review by kirillsud — COMMENTED** [review]
Great example! I have no critical comments about the implementation. The code is clear and focused on the scenario. I have only small comments like using `console.log` and etc.

File: examples/ai-image-generator/pages/index.jsx (hunk 2 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx:25]: yes good point - don't need this.

File: examples/ai-image-generator/pages/index.jsx (hunk 4 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx:86]: great point - thank you for this!

**Comment by horeaporutiu** [comment]
Should be ready for app-examples (hopefully 🤞🏼) now!

File: examples/ai-image-generator/pages/index.jsx (hunk 4 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx:73]: Ah - sorry still need to update this one, just one second @kirillsud

Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx:73]: ok should be ready now

File: examples/ai-image-generator/pages/index.jsx [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        authUrl: miro.getAuthUrl(),
+      },
+    };
+  }
+  return {
+    props: {},
+  };
+};
+
+export default function Main({ boards, authUrl }) {
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/index.jsx:22]: 'boards' is defined but never used

File: examples/ai-image-generator/pages/index.jsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  const [inputValue, setInputValue] = useState("");
+
+  //handles the prompt input being typed in
+  const handleInputChange = (newValue) => {
+    setInputValue(newValue);
+  };
+
+  // Shows spinner while API calls are in progress / image is being dragged & dropped
+  const showSpinner = async () => {
+    let spinner = await document.getElementById("spinner");
```
Comment (kirillsud) [inline examples/ai-image-generator/pages/index.jsx:63]: `getElementById` is the sync method:

```suggestion
  const showSpinner = () => {
    const spinner = document.getElementById("spinner");
```

Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx]: Thank you - just changed

File: examples/ai-image-generator/pages/index.jsx (hunk 5 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/pages/index.jsx]: just changed, thank you!

File: examples/ai-image-generator/components/Button.jsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
+// components/Button.js
```
Comment (mettin) [inline examples/ai-image-generator/components/Button.jsx:1]: can we remove 
```suggestion

```

File: examples/ai-image-generator/pages/index.jsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
+import React, { useState, useEffect } from "react";
+import initMiro from "../initMiro";
+//React components for the prompt and button
```
Comment (mettin) [inline examples/ai-image-generator/pages/index.jsx:3]: Remove comment

File: examples/ai-image-generator/src/assets/style.css [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+  display: flex;
+}
+
+#root {
+  overflow: auto;
+  padding: var(--space-medium);
+}
+
+.button {
+  width: 300px;
```
Comment (mettin) [inline examples/ai-image-generator/src/assets/style.css:25]: 100%?

File: examples/ai-image-generator/src/assets/style.css [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  object-fit: cover; /* Ensure the image covers the entire container while maintaining its aspect ratio */
+  background: url(https://encrypted-tbn2.gstatic.com/images?q=tbn:ANd9GcRvF4WdZJSA4MkWJXClae4eCvSdk87c5Ok63fgxBxVyR6aHB2Ju_A)
+    no-repeat center;
+}
+
+.spinner {
+  position: fixed;
+  height: 2em;
+  width: 2em;
+  border: 3px solid rgba(255, 255, 255, 0.3);
```
Comment (mettin) [inline examples/ai-image-generator/src/assets/style.css:51]: Can we use mirotone vars here?

Comment (horeaporutiu) [reply examples/ai-image-generator/src/assets/style.css:51]: went through the Mirotone docs but not sure which ones you are referring to here? Which one specifically?

File: examples/ai-image-generator/src/assets/style.css (hunk 9 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/src/assets/style.css:19]: Please let me know if this is a blocker - then I can maybe take more time to implement this with cs1 ce12 syntax. if it's not a blocker will make video / merge as is then we can make a PR to update this in the future

File: examples/ai-image-generator/src/assets/style.css (hunk 10 above)
Comment (horeaporutiu) [reply examples/ai-image-generator/src/assets/style.css]: added.

File: examples/ai-image-generator/src/assets/style.css (hunk 9 above)
Comment (mettin) [reply examples/ai-image-generator/src/assets/style.css:19]: Fine, keep as is! styling isnt the point of this demo

File: examples/ai-image-generator/pages/index.jsx [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+  //drag and drop logic
+  const drop = async (e) => {
+    showSpinner();
+    const { x, y, target } = e;
+    if (target instanceof HTMLImageElement) {
+      const image = await window.miro.board.createImage({
+        x,
+        y,
+        url: target.src,
+      });
```
Comment (horeaporutiu) [inline examples/ai-image-generator/pages/index.jsx:57]: Can use image.setMetadata here

File: examples/ai-image-generator/pages/api/openai.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+export default async (req, res) => {
+  // grab prompt from the front end
+  let prompt = await req.body.prompt;
+
+  try {
+    const response = await openai.images.generate({
+      prompt: prompt,
+    });
+    let image_url = response.data[0].url;
```
Comment (mettin) [inline examples/ai-image-generator/pages/api/openai.js:15]:
```suggestion
    const image_url = response.data[0].url;
```

**Review by mettin — APPROVED** [review]

**Pull request merged**